lib/sqlalchemy/sql/selectable.py

```diff
@@ -4548,31 +4576,34 @@ def _column_naming_convention(
         pa = prefix_anon_map()
         names = set()
 
-        def go(c, col_name=None):
+        def go(
+            c: Union[ColumnElement[Any], TextClause],
+            col_name: Optional[str] = None,
+        ) -> Optional[str]:
             if c._is_text_clause:
                 return None
 
             elif not dedupe:
-                name = c._proxy_key
+                name = c._proxy_key  # type: ignore
                 if name is None:
                     name = "_no_label"
                 return name
 
```

Comment:
**mike bayer** (zzzeek) wrote:

remove

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4325